src/main/java/com/snowflake/kafka/connector/Utils.java

```diff
@@ -137,6 +137,18 @@ public class Utils {
 
   private static final KCLogger LOGGER = new KCLogger(Utils.class.getName());
 
+  /**
+   * Quote the column name if needed: When there is quote already, we do nothing; otherwise we
+   * convert the name to upper case and add quote
+   */
+  public static String quoteNameIfNeeded(String name) {
+    int length = name.length();
+    if (name.charAt(0) == '"' && (length >= 2 && name.charAt(length - 1) == '"')) {
```

Comment:
can use startswith and endswith?

Reply:
No strong preference here, but charAt should performance better than startswith? I see startswith is usually used for matching a string instead of a character